tests/test_data_management.py

```diff
@@ -161,7 +161,9 @@ class TestPlatformSerializationWithRuamelYaml:
     """Unit tests to check the serialization is keeping all we want to keep."""
 
     original_platform = ql.build_platform(Galadriel.runcard)
-    saved_platform = ql.build_platform(save_platform(path="./test.yml", platform=original_platform))
+    path = save_platform(path="./test.yml", platform=original_platform)
+    saved_platform = ql.build_platform(path)
+    os.remove(path)
```

Comment:
Instead of creating and removing a file, why not using `YAML().dump` to dump the information into a stream, and then retrieve it from there?

[See this link for more information.](https://stackoverflow.com/questions/61722242/dump-the-yaml-to-a-variable-instead-of-streaming-it-in-stdout-using-ruamel-yaml)

Reply:
Yess, that what I said yesterday that I was gonna do! Thanks!!!

I think I have actually done that already above in this discussion, in the console image I pasted! 🙌

Reply:
@AlbertMitjans I don't see the solution that straight forward, now that I have looked it better... 😬 

Because I wanna test this function:
![Screenshot 2023-10-27 at 21 03 15](https://github.com/qilimanjaro-tech/qililab/assets/109400222/be1d90b6-38c3-4376-9c16-a7ca881f309a)
so I would need to patch in some way the with open, so that it doesn't open a file and saves it into a variable...

Notice I cannot just do what you said, because I would then just not be testing this function that we wanna test 😅

Reply:
The `save_platform` function is already tested in this file. As you mention in the comments, the tests you added are to make sure ruamel works properly. Thus I see no need of using `save_platform`.

I would suggest using `YAML().dump` directly to obtain the `saved_platform` variable, and then run the tests.

Reply:
I mean, it makes sense, but I don't know if I completely agree on that. 🤔 

Since the idea I had for this test, was testing that both methods work together properly, kind of like an integration test of both functions. 

What you are proposing is an extended unit test for the `build_platform`, working correctly with the YAML dump function. 

And if in the future someone changes the `save_platform` but not the `build_platform` one, then the test would still pass, when maybe it shouldn't? 🤔  

What do you think? Still, think it's better to not test the integration of both methods?

Reply:
Since this conversation is going too long, for a single test, I'll do as you say in the next answer hehe 🙃

Reply:
Check my new comment. If you want to test that `save_platform` saves the exact same dictionary, I believe one test would suffice, meaning:

```python
original_platform = ql.build_platform(Galadriel.runcard)
path = save_platform(path="./test.yml", platform=original_platform)
saved_platform = ql.build_platform(path)
os.remove(path)
assert original_platform == saved_platform
```

I would still do this test without using `save_platform`:

```python
original_platform = Galadriel.runcard
saved_platform = YAML().dump(original_platform.to_dict(), stream=...)
assert original_platform == saved_platform
```

Reply:
Bueno, en el primer caldria possar un `to_dict` por ahi, porque sino les dues classes de Platform no seran les mismes

Reply:
Please Guille, use english so everyone in the team can follow up on these discussions.

Reply:
Yesyes, I actually, had deleted the comment already!

Reply:
In the end, as we commented, I ended up reducing the tests, by joining them with the already existing one very in few lines 👍